panel/widgets/slider.py

```diff
@@ -338,14 +350,19 @@ def _get_embed_state(self, root, values=None, max_opts=3):
                              "in the %s widgets' values list." % type(self).__name__)
         return self, model, values, lambda x: x.value, 'value', 'cb_obj.value'
 
+    @staticmethod
+    def _format_value(value, formatter):
+        if formatter.startswith("%"):
+            return formatter % value
+        return f'{{:{formatter}}}'.format(value)
+
     @property
     def labels(self):
         title = (self.name + ': ' if self.name else '')
-        if isinstance(self.options, dict):
-            return [title + ('<b>%s</b>' % o) for o in self.options]
-        else:
-            return [title + ('<b>%s</b>' % (o if isinstance(o, string_types) else (self.formatter % o)))
-                    for o in self.options]
+        return [title + ('<b>%s</b>' % (o if isinstance(o, string_types)
+            else self._format_value(o, self.formatter)))
+            for o in self.options]
```

Comment:
This does not seem correct to me, if you're providing an `options` dictionary then the keys are the formatted values and don't need to have further formatting applied.

Reply:
Also what happens if the values are neither strings or floats/ints? You can have any object type in the `options`.